x/concentrated-liquidity/pool_store.go

```diff
@@ -1,22 +1,46 @@
 package concentrated_liquidity
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/osmosis-labs/osmosis/v12/osmoutils"
 	types "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
 )
 
-func (k Keeper) getPoolbyId(ctx sdk.Context, poolId uint64) Pool {
+func (k Keeper) MarshalPool(pool types.PoolI) ([]byte, error) {
```

Comment:
does this need to be exported?

Reply:
I don't see why not, but still made it a private method